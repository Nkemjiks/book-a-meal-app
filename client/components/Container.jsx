import React from 'react';
import { createBrowserHistory } from 'history';
import { Router, Route, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';

import store from '../store';
import Routes from './Routes';

const history = createBrowserHistory();

/**
 * landing page of the application
 *
 * @description React Component encapsulating application user interface
 *
 * @returns {JSX} JSX representation of component
 */
const Container = () => {
  return (
    <Provider store={store}>
      <Router history={history}>
        <Switch>
          <Route path="/" component={Routes} />
        </Switch>
      </Router>
    </Provider>
  );
};

export default Container;
