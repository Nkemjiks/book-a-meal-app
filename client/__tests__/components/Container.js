import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Container from '../../components/Container';

Enzyme.configure({ adapter: new Adapter() });

describe('Container Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders page Route', () => {
    const wrapper = shallow(<Container />);
    expect((wrapper).find('Provider').length).toBe(1);
    expect((wrapper).find('Router').length).toBe(1);
    expect((wrapper).find('Switch').length).toBe(1);
    expect((wrapper).find('Route').length).toBe(1);
  });
});
