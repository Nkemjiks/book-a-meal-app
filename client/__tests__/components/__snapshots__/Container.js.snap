// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container Component renders correctly 1`] = `null`;
